internal/services/cdn/cdn_frontdoor_custom_domain_association_resource_test.go

```diff
@@ -0,0 +1,242 @@
+package cdn_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type CdnFrontDoorCustomDomainAssociationResource struct {
+}
+
+// NOTE: There isn't a complete test case because the basic and the
+// update together equals what the complete test case would be...
+func TestAccCdnFrontDoorCustomDomainAssociation_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cdn_frontdoor_custom_domain_association", "test")
+	r := CdnFrontDoorCustomDomainAssociationResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+	})
+}
+
+// NOTE: the 'requiresImport' test is not possible on this resource
```

Comment:
It currently already validates if the custom domain belongs to the route or not and I return the below error, are you suggesting I throw the import error here instead of my own error?

```
return out, fmt.Errorf("the CDN FrontDoor Route(Name: %q) is currently not associated with the CDN FrontDoor Custom Domain(Name: %q). Please remove the CDN FrontDoor Route from your 'cdn_frontdoor_custom_domain_association' configuration block", v.RouteName, id.CustomDomainName)
```